compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -239,23 +239,20 @@ object Implicits:
         Nil
       else
         val nestedCtx = ctx.fresh.addMode(Mode.TypevarsMissContext)
+        val candidates = new mutable.ListBuffer[Candidate]
+        var extensionOnly = true
```

Comment:
I get that using a var here means that the `tryCandidate` closure can be reused, but the trade-off is that this var will become a BooleanRef and thus be allocated too, so I don't think we're gaining anything. Maybe instead `extensionOnly` and `tryCandidate` could be put in an object.

Reply:
You are right. I think it's fine to use two closures instead.